nexus_constructor/InstrumentView.py

```diff
@@ -0,0 +1,260 @@
+        light_blue = QColor("lightblue")
+        dark_red = QColor("#b00")
+
+        self.set_material_properties(self.grey_material, black, grey)
+        self.set_material_properties(self.red_material, red, dark_red)
+        self.set_material_properties(self.green_material, grey, grey)
+        self.set_material_properties(self.beam_material, blue, light_blue, 0.5)
+
+    @staticmethod
+    def set_cube_mesh_dimensions(cube_mesh, x, y, z):
+        """
+        Sets the dimensions of a cube mesh.
+        :param cube_mesh: The cube mesh to modify.
+        :param x: The desired x extent.
+        :param y: The desired y extent.
+        :param z: The desired z extent.
+        """
+        cube_mesh.setXExtent(x)
+        cube_mesh.setYExtent(y)
+        cube_mesh.setZExtent(z)
+
+    def setup_sample_cube(self):
+        """
+        Sets up the cube that represents a sample in the 3D view by giving the cube entity a mesh and a material.
+        """
+        self.set_cube_mesh_dimensions(self.cube_mesh, *self.sample_cube_dimensions)
+        self.add_components_to_entity(
+            self.cube_entity, [self.cube_mesh, self.red_material]
+        )
+
+    @staticmethod
+    def add_components_to_entity(entity, components):
+        """
+        Takes a QEntity and gives it all of the components that are contained in a list.
+        """
+        for component in components:
+            entity.addComponent(component)
+
+    @staticmethod
+    def set_cylinder_mesh_dimensions(cylinder_mesh, radius, length, rings):
+        """
+        Sets the dimensions of a cylinder mesh.
+        :param cylinder_mesh: The cylinder mesh to modify.
+        :param radius: The desired radius.
+        :param length: The desired length.
+        :param rings: The desired number of rings.
+        """
+        cylinder_mesh.setRadius(radius)
+        cylinder_mesh.setLength(length)
+        cylinder_mesh.setRings(rings)
+
+    @staticmethod
+    def set_beam_transform(cylinder_transform):
+        """
+        Configures the transform for the beam cylinder by giving it a matrix.
+        :param cylinder_transform: A QTransform object.
+        """
+        cylinder_matrix = QMatrix4x4()
+        cylinder_matrix.rotate(270, QVector3D(1, 0, 0))
+        cylinder_matrix.translate(QVector3D(0, 20, 0))
+
+        cylinder_transform.setMatrix(cylinder_matrix)
+
+    def setup_beam_cylinder(self):
+        """
+        Sets up the beam cylinder by giving the cylinder entity a mesh, a material, and a transformation.
+        """
+        self.set_cylinder_mesh_dimensions(
+            self.cylinder_mesh, 2.5, self.cylinder_length, 2
+        )
+        self.set_beam_transform(self.cylinder_transform)
+        self.add_components_to_entity(
+            self.cylinder_entity,
+            [self.cylinder_mesh, self.beam_material, self.cylinder_transform],
+        )
+
+    @staticmethod
+    def set_sphere_mesh_radius(sphere_mesh, radius):
```

Comment:
pass in a mock to these tests and check that `setRadius` has been called here for example.